### ISSUE TITLE
Legacy fact check should detect/remediate unscoped legacy facts.

### ISSUE THREAD

**Issue by binford2k** [puppetlabs/puppet-lint#129]
This [spec test](https://github.com/puppetlabs/puppet-lint/blob/cc275ca0da4a7589b3194790d7e6353595e54499/spec/unit/puppet-lint/plugins/legacy_facts/legacy_facts_spec.rb#L21-L27) validates that an unscoped fact like `$osfamily` is not detected or remediated. I think that it should be detected, especially in context of Puppet 8 deprecating those facts.

This means that people writing code that shadows those legacy facts by using the same variable name will now trigger a warning. If they genuinely want to do that, then they can configure the linter to skip that code 😁 

@joshcooper @bastelfreak

**Comment by alexjfisher** [comment]
Already picked up by http://puppet-lint.com/checks/variable_scope/ surely?

**Comment by binford2k** [comment]
@alexjfisher I think that it should still correct this. If the user dislikes that style and disables that check, the legacy fact check should still identify known obsoleted facts.

**Comment by fbhdk** [comment]
I agree that legacy_facts should detect this, also in order for --fix to correct these. It will save a lot of people a lot of time.
